feat(metrics): Expose span.duration

narsaynorath opened the pull request:
Exposes span.duration in the metrics query builder with a very limited set of available tags and replaces the `self_time` and `exclusive_time` metrics. We've decided that we do not want to
show these anymore.

People who have built widgets using this dashboard will still work, but they won't be able to select the field in new widgets, or re-select it in modified widgets.

narsaynorath committed:
feat(metrics): Expose span.duration

Exposes span.duration in the metrics query builder with a very limited
set of available tags.

narsaynorath commented:
If we want to move forward with this, should this be feature flagged? 🤔

phacops commented:
> If we want to move forward with this, should this be feature flagged? 🤔

I don't think so, we do not want `exclusive_time` exposed for the release.

narsaynorath committed:
Expand allowed ops

narsaynorath committed:
Remove self time metric

narsaynorath committed:
Add default to tag data

narsaynorath committed:
Use one filter call

So I don't have to ensure there's a default fallback or loop over the
list of tags multiple times

narsaynorath committed:
Fix comment
